src/lib/containers/oauth/OAuthManagement.tsx

```diff
@@ -5,19 +5,43 @@ import moment from 'moment'
 import { useGetOAuthClientInfinite } from '../../utils/hooks/SynapseAPI'
 import { CreateOAuthModal } from './CreateOAuthClient'
 import { OAuthClient } from '../../utils/synapseTypes/OAuthClient'
+import WarningModal from '../synapse_form_wrapper/WarningModal'
+import { SynapseClient } from '../../utils'
+import { useSynapseContext } from '../../utils/SynapseContext'
 import Typography from '../../utils/typography/Typography'
+import CopyToClipboardInput from '../CopyToClipboardInput'
 
 export const OAuthManagement: React.FunctionComponent = () => {
+  const { accessToken } = useSynapseContext()
   const [isShowingCreateClientModal, setIsShowingCreateClientModal] =
     useState(false)
   const [isEdit, setIsEdit] = useState<boolean>(false)
   const [selectedClient, setSelectedClient] = useState<OAuthClient>()
   const [isShowingConfirmModal, setIsShowingConfirmModal] = useState(false)
+  const [isShowingSecretWarning, setIsShowingSecretWarning] = useState(false)
+  const [isShowingSecret, setIsShowingSecret] = useState(false)
+  const [secret, setSecret] = useState<string>()
   const [isShowingVerification, setIsShowingVerification] = useState(false)
 
   const { data, hasNextPage, fetchNextPage } = useGetOAuthClientInfinite()
   const oAuthClientList = data?.pages.flatMap(page => page.results) ?? []
 
+  const warningHeader = 'Are you absolutely sure?'
+  const warningBody =
+    'If you have an existing secret, generating a new secret will make your application invalid after generation. This action cannot be undone.'
+
+  const onShowSecret = async () => {
+    setIsShowingSecretWarning(false)
+    setSelectedClient(undefined)
+    setIsShowingSecret(true)
+
+    const secret = await SynapseClient.createOAuthClientSecret(
```

Comment:
What happens if this call fails?  Since this is not in a try/catch, I'm concerned that the error is lost (does this propagate to an error boundary?)

Reply:
At the very least, I'd like to see a displayToast in a catch block so an error is not lost